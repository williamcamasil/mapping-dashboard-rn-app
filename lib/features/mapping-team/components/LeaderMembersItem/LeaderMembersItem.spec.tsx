import React from 'react';

import {
  render, waitFor,
} from '@testing-library/react-native';
import { ModalProvider } from 'mapping-style-guide-rn';

import LeaderMembersItem from '.';

const onCardPressMock = jest.fn();
const onContactDeletedMock = jest.fn();

const renderComponent = async (cardData: any) => {
  const component = render(
    <ModalProvider>
      <LeaderMembersItem
        itemData={cardData}
        onItemPress={onCardPressMock}
        onItemDeleted={onContactDeletedMock}
        isAdminUser
      />
    </ModalProvider>,
  );
  await waitFor(() => component.UNSAFE_getByType(LeaderMembersItem));
  return component;
};

const favoredDataMock = {};

describe('LeaderMembersItem', () => {
  it('Should match snapshot', async () => {
    const component = await renderComponent(favoredDataMock);
    expect(component).toMatchSnapshot();
  });
});
