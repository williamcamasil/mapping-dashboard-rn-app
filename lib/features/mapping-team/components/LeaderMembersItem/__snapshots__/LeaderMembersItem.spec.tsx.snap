// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderMembersItem Should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingRight": 12,
    }
  }
  testID="item-list"
>
  <View
    style={
      {
        "flex": 1,
        "maxWidth": "88%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      />
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "height": 4,
          "width": 4,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#676D79",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    />
  </View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    onPress={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    responsible={true}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    testID="icon-dots"
    vbHeight={32}
    vbWidth={32}
    width={24}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        name="Layer_2"
      >
        <RNSVGPath
          d="M16 7a2 2 0 112-2 2 2 0 01-2 2zm0-2zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zM16 18a2 2 0 112-2 2 2 0 01-2 2zm0-2zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zM16 29a2 2 0 112-2 2 2 0 01-2 2zm0-2zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "strokeWidth",
            ]
          }
          strokeWidth="1.8"
        />
      </RNSVGGroup>
      <RNSVGPath
        d="M0 0H32V32H0z"
        fill={null}
        name="frame"
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
